coap-core/src/main/java/com/mbed/coap/transport/udp/DatagramSocketTransport.java

```diff
@@ -160,13 +135,16 @@ public void sendPacket0(CoapPacket coapPacket, InetSocketAddress adr, TransportC
         socket.send(datagramPacket);
     }
 
-    @Override
+    protected boolean socketCreated() {
+        return (socket != null);
+    }
+
     public InetSocketAddress getLocalSocketAddress() {
-        return (InetSocketAddress) socket.getLocalSocketAddress();
+        return (InetSocketAddress) ((DatagramSocketAdapter) getSocket()).getLocalSocketAddress();
```

Comment:
How this will work when socket is other type than `DatagramSocketAdapter`?

Reply:
Fixed.